### PR TITLE
Paginate build list.

### PR DESCRIPTION
**Pull request by n1k0**
Note: this is not intended for landing as is. This patch illustrates how the new elm-kinto pagination system works, with a working implementation.

We'll definitely wait for #40 to land first before landing this, especially as this can't fully work with in-memory filtering.

### CODE REVIEW

File: ui/Model.elm [hunk 1]
```diff
@@ -27,10 +27,21 @@ getBuildRecordList : Cmd Msg
 getBuildRecordList =
     client
         |> Kinto.getList recordResource
+        |> Kinto.limit 10
```
Comment (magopian) [inline ui/Model.elm]: should we put something like 100 here? It seems filtering 100 objects in memory (if we'll keep doing that?) shouldn't be too memory/cpu intensive

Comment (n1k0) [inline ui/Model.elm]: :+1:

**Comment by magopian** [comment]
I believe this is now read to review, @n1k0 @Natim ?

File: ui/Main.elm [hunk 2]
```diff
@@ -32,6 +38,29 @@ update msg ({ filterValues } as model) =
...
+                        | buildsPager = newPager
+                        , filteredBuilds = newPager.objects
+                        , loading = False
+                    }
+                    ! []
+
+        BuildRecordsNextPageFetched (Err err) ->
+            let
+                _ =
+                    Debug.log "An error occured while fetching the next page of build records" err
```
Comment (n1k0) [inline ui/Main.elm]: Nit: We should file an issue to have friendly error messages in the UI.

File: ui/Main.elm [hunk 3]
```diff
@@ -21,8 +22,13 @@ main =
 update : Msg -> Model -> ( Model, Cmd Msg )
 update msg ({ filterValues } as model) =
     case msg of
-        BuildRecordsFetched (Ok buildRecordList) ->
-            updateModelWithFilters ({ model | builds = buildRecordList, loading = False })
+        BuildRecordsFetched (Ok buildsPager) ->
+            updateModelWithFilters
+                { model
+                    | buildsPager = buildsPager
+                    , filteredBuilds = buildsPager.objects
```
Comment (n1k0) [inline ui/Main.elm]: I don't think this is actually used anymore, and should be removed.

Comment (magopian) [inline ui/Main.elm]: Very well spotted, thanks!
And pulling that string, it came with a bunch of other dead code <3

File: ui/Main.elm (hunk 2 above)
Comment (magopian) [inline ui/Main.elm]: :+1: #79

**Pull request merged**